### PR TITLE
HHH-14249 MultiLineImport fails when script contains blank spaces or tabs at the end of the last sql statement

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-14249

### CODE REVIEW

File: hibernate-core/src/main/antlr/sql-script.g [hunk 1]
```diff
@@ -57,16 +57,23 @@ options {
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 script
-	: (newLineToSkip)* (statement)* EOF
+	: (newLineToSkip)* (SPACE)* (TAB)* (statement)* EOF
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/antlr/sql-script.g:60]: would the above rule work for `(TAB) * (SPACE)*`, btw?

Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-script.g:60]: Good catch, I'm going to fix it.

Thanks @NathanQingyangXu

Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-script.g]: It should be fixed now, Thanks @NathanQingyangXu

File: hibernate-core/src/main/antlr/sql-script.g [hunk 2]
```diff
@@ -57,16 +57,23 @@ options {
...
 
 script
-	: (newLineToSkip)* (statement)* EOF
+	: (newLineToSkip | SPACE | TAB )* (statement)* EOF
     ;
 
 statement
-	: { statementStarted(); } (statementPart (afterStatementPartNewline)*)* DELIMITER (newLineToSkip)* { statementEnded(); }
+	: { statementStarted(); } statementFirstPart (statementPart (afterStatementPartNewline)*)* DELIMITER (SPACE | TAB | newLineToSkip)* { statementEnded(); }
+	;
```
Comment (Sanne) [inline hibernate-core/src/main/antlr/sql-script.g:65]: maybe you could reorganize the "whitespace" into its own group to make things simpler?

```
script
	: ignoredWhitespace (statement ignoredWhitespace)* EOF
    ;

statement
	: { statementStarted(); } statementFirstPart (statementPart (afterStatementPartNewline)*)* DELIMITER { statementEnded(); }
	;

ignoredWhitespace
	: (newLineToSkip | SPACE | TAB )*
	;
```

Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-script.g:65]: Thanks @Sanne I made the suggested change.

**Pull request merged**